shared/js/media/crop_resize_rotate.js

```diff
@@ -488,8 +498,48 @@ function cropResizeRotate(blob, cropRegion, outputSize, outputType,
         URL.revokeObjectURL(baseURL);
       }
 
+      var inputBlob = blob; // XXX just use a different parameter name
+
       // Finally, encode the image into a blob
-      canvas.toBlob(gotEncodedBlob, outputType || JPEG);
+
+      var postCallback = preserveExif ? copyMetadata : gotEncodedBlob;
+
+      canvas.toBlob(postCallback, !outputType ? JPEG : outputType);
```

Comment:
For the second argument, `outputType || JPEG` seems clearer to me. Isn't that the same as your conditional operator?